spec/basket_spec.rb

```diff
@@ -2,19 +2,35 @@
 
 RSpec.describe Basket do
 
-  it "creates basket" do
-    expect Basket.new.is_a? Basket
+  let(:product1) { Product.new(name: "Football", price: 79) }
+  let(:product2) { Product.new(name: "Basketball", price: 60) }
+  let(:basket1) { Basket.new([product1, product2]) }
+
+  it "calculates products sum correctly" do
+    expect(basket1.sum).to eq 139
   end
 
-  let(:p1) { Product.new(name: "Football", price: 79) }
-  let(:p2) { Product.new(name: "Basketball", price: 60) }
   it "checks if sum with vat is correct" do
-    sum = Basket.new([p1, p2]).sum
-    sum_with_vat = Basket.new([p1, p2]).sum_with_vat
-    expect(sum * 1.23).to eq sum_with_vat
+    sum = Basket.new([product1, product2]).sum
+    sum_with_vat = Basket.new([product1, product2]).sum_with_vat
+    expect(sum * 1.23).to eq (sum_with_vat)
   end
 
+  describe "#delete" do
+    context "for product not existing in basket" do 
+      it "raies ProductNotFound error" do
+        expect {
+          basket1.delete(0)
+          }.to raise_error(Basket::ProductNotFound)
+      end
+    end
 
-
-  #it "doesn't take basket arguments"
+    context "for product existing in basket" do
+      it "deletes product from basket" do
+        prod = Product.new(name: "Product", price: 50)
+        bas = Basket.new([prod])
```

Comment:
you have access to `basket1` and `product1` here. Why you can't re-use them here?